simple-loadflow/src/main/java/com/powsybl/loadflow/simple/equations/IndexedNetwork.java

```diff
@@ -27,10 +26,54 @@
 
     private final Map<Bus, Integer> index;
 
+    static class Neighbor {
+
+        final Branch branch;
+        final Branch.Side side;
+
+        Neighbor(Branch branch, Branch.Side side) {
+            this.branch = branch;
+            this.side = side;
+        }
+
+        @Override
+        public String toString() {
+            return "Neighbor(branchId='" + branch.getId() + "', side=" + side + ")";
+        }
+    }
+
+    private final List<Branch> branches = new ArrayList<>();
+
+    private final SortedMap<Integer, SortedMap<Integer, List<Neighbor>>> neighbors = new TreeMap<>();
+
     public IndexedNetwork(Network network) {
         this.network = Objects.requireNonNull(network);
-        buses = network.getBusView().getBusStream().collect(ImmutableList.toImmutableList());
+        buses = network.getBusView().getBusStream().filter(Bus::isInMainConnectedComponent).collect(ImmutableList.toImmutableList());
         index = IntStream.range(0, buses.size()).boxed().collect(ImmutableMap.toImmutableMap(i -> buses.get((int) i), i -> i));
+        for (Branch br : network.getBranches()) {
+            Bus bus1 = br.getTerminal1().getBusView().getBus();
+            Bus bus2 = br.getTerminal2().getBusView().getBus();
+            if (bus1 != null && bus1.isInMainConnectedComponent() && bus2 != null && bus2.isInMainConnectedComponent()) {
+                branches.add(br);
+
+                int busNum1 = index.get(bus1);
+                int busNum2 = index.get(bus2);
+
```

Comment:
An afterthought after trying to write documentation about iteration on branches:
the structure with Neigbours feels a little weird, because for example we introduce for each bus itself as a neighbour.

Actually what we are building is not really a list of neighbours, but a list of dependent variables.
Here we put bus1 as a neighbour of bus1, but what we really mean is that injection at bus 1 (which is at row "busNum1" in the rhs of the mathematical problem) depends on the phase at bus 1 (which is the variable at column "busNum1" in the mathematical problem).

I think it will be important to make that distinction for the building of the AC matrix, for which we will have 2 lines/columns per bus.

So the structure is more a map (numTarget, numVariable) --> branch terminal. Maybe we should just rename it and the "forEachBranchInCscOrder" to make it clear. Still need to find a name though :)

Reply:
Agreed, it is not a neighborhood structure but a functions/variables dependency structure convenient for jacobian building.
I totally refactored this, in "jac" branch to support functions with multiples derivative variables where it is not possible to directly map bus num and variables.

Reply:
Ok. I think that the more relevant structure may be to have a sparse matrix which elements would be the equations used to compute the actual matrix. It would allow to create "equations" object only once, too, when creating that matrix.
(this is what the map of map is representing here)

Reply:
But you need to define link between bus -> equation -> variable -> partial derivative (static method) -> (row, col).
This is what is done partly in IndexedNetwork.createJacobian and also in BranchHandler (badly named) subclasses for a specific equiation set.
Do you have in mind a design that would also allow to easily and safely update code when it comes to add a new feature in jacobian (like another variable to implement phase shifter control for instance) and that would not force developper to rethink about how matrix/network layout is done?